station-world-events-v0/src/main/java/net/modificationstation/stationapi/api/celestial/CelestialEvent.java

```diff
@@ -0,0 +1,270 @@
+        }
+        if (active) {
+            activityState.active = true;
+            activityState.attemptedActivation = true;
+            activityState.markDirty();
+            onActivation();
+        }
+        return active;
+    }
+
+    /**
+     * Called precisely on event activation.
+     * Can be customized by inheriting from this class.
+     * Has access to the world object for more versatility.
+     */
+    public void onActivation() {
+    }
+
+    /**
+     * Called precisely on event deactivation.
+     * Can be customized by inheriting from this class.
+     * Has access to the world object for more versatility.
+     */
+    public void onDeactivation() {
+    }
+
+    /**
+     * Updates the activity status of an already active event.
+     * Used by CelestialTimeManager 4 times per day.
+     * Checks for validity of the time interval but not for chance and incompatibility.
+     *
+     * @param worldTime Current time in the world measured in ticks.
+     */
+    public void updateEvent(long worldTime) {
+        CelestialEventActivityState activityState = (CelestialEventActivityState) this.world.getOrCreateState(CelestialEventActivityState.class, this.name);
+        if (initializationNeeded) {
+            activityState = initializeEvent(activityState);
+        }
+        if (!active && !activityState.active) return;
+        worldTime -= startingDaytime;
+        worldTime += endingDaytime;
+        long days = (worldTime / dayLength) + dayOffset;
+        active = days % frequency <= extraDays;
+        activityState.active = active;
+        if (!active) {
+            onDeactivation();
+            activityState.attemptedActivation = false;
+        }
+        activityState.markDirty();
+    }
+
+    /**
+     * Initializes the event by synchronizing it with NBT data.
+     * Only used internally.
+     *
+     * @param activityState CelestialEventActivityState (custom PersistentState) used for accessing NBT data.
+     * @return The initialized CelestialEventActivityState.
+     */
+    private CelestialEventActivityState initializeEvent(CelestialEventActivityState activityState) {
+        initializationNeeded = false;
+        activityState = (CelestialEventActivityState) this.world.getOrCreateState(CelestialEventActivityState.class, this.name);
+        if (activityState == null) {
+            activityState = new CelestialEventActivityState(this.name);
+            this.world.setState(this.name, activityState);
+        } else {
+            active = activityState.active;
+        }
+        return activityState;
+    }
+
+    /**
+     * Manual way of stopping event.
+     * Intended for command systems.
+     */
+    public void stopEvent() {
+        onDeactivation();
+        if (active) {
+            CelestialEventActivityState activityState = (CelestialEventActivityState) this.world.getOrCreateState(CelestialEventActivityState.class, this.name);
```

Comment:
Celestial events should probably be stored in the same state file instead of multiple files for each event. Something like stationapi_celestial_events.dat

Reply:
I am not very experienced with file saving, so I might need some help to figure out how to do this change